STYLE_GUIDE.md

```diff
@@ -0,0 +1,48 @@
+# Style guide
+
+This style guide defines specific coding standards and advice for this Java codebase. The rules here are extensions to the [Google Java Style Guide](https://google.github.io/styleguide/javaguide.html).
+
+Please see the [contributing guide](CONTRIBUTING.md) for general guidance for contributing to this project.
+
+### Automatic formatting
+
+Automatic formatting should be performed with `./gradlew goJF` or `./mvnw fmt:format`. Formatting all projects can be done with `./build.sh format`.
+
+### Class member order
+
+*Extends [3.4.2](https://google.github.io/styleguide/javaguide.html#s3.4.2-ordering-class-contents)*
+
+Class members should be in the following order, in decreasing priority:
+
+1. Static before non-static
```

Comment:
While I generally agree with tending to order things in more or less this way, I've found the more relaxed logical ordering requirement in the linked google style guide to work pretty well.  In a way, this section goes against the advice in the Google style-guide.

Defining symbols close to first use is a common practice at Google that often helps readability but would be disallowed by this strict ordering.